src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs

```diff
@@ -602,7 +602,7 @@ private void WriteByteSlow(byte value)
                 ClearReadBufferBeforeWrite();
                 EnsureBufferAllocated();
             }
-            else if (_writePos == _bufferSize - 1)
+            else
             {
                 FlushWrite();
```

Comment:
So this method should only be called when either _writePos == 0 or _writePos == _bufferSize, right? 
Consider adding the assertion, but not in this PR since we want to merge it ASAP.

```suggestion
                Debug.Assert(_writePos == _bufferSize);
                FlushWrite();
```